File: Jessica/83.Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:        
        node=head
        while node:
            while node.next and node.val ==node.next.val:
                node.next=node.next.next
            node=node.next
        return head

File: Jessica/83.Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
       ListNode* tmp=head;
        while(tmp){
            while(tmp->next && tmp->next->val== tmp->val)
                tmp->next=tmp->next->next;
            tmp=tmp->next;
            
        }
        return head;
    }
};